packages/go_router/lib/src/route.dart

```diff
@@ -432,16 +430,15 @@ class GoRoute extends RouteBase {
 
   // TODO(chunhtai): move all regex related help methods to path_utils.dart.
   /// Match this route against a location.
-  RegExpMatch? matchPatternAsPrefix(String loc) =>
-      _pathRE.matchAsPrefix(loc) as RegExpMatch?;
+  RegExpMatch? matchPatternAsPrefix(String path) => pattern.match(path);
 
   /// Extract the path parameters from a match.
   Map<String, String> extractPathParams(RegExpMatch match) =>
-      extractPathParameters(pathParameters, match);
+      pattern.extractPathParameters(match);
 
   /// The path parameters in this route.
   @internal
-  final List<String> pathParameters = <String>[];
+  List<String> get pathParameters => pattern.parameters;
```

Comment:
Not sure if the pattern should be accessed directly (`route.pattern.parameters`) or if the route should facade to the pattern.

Reply:
if we expose the pattern, than they should probably be accessed directly to avoid duplication

Reply:
do we want to expose pattern though?

Reply:
Currently no, the idea was to have `RoutePattern` used internally, (I have to make it internal in the future) and have the public representation still `String path`, ie: `GoRoute(path: '/some-path')`.  The `RoutePattern` is supposed to be used internally everywhere so we don't end up with `String path`, where we are unsure whether path is `/books/:bookId` or `/books/123` without looking at the wider context.

However I'm not a fan of this kind of constructor conversion, it's a bit harder to reason about.

Reply:
in that case the RoutePattern should be kept in as private variable in GoRoute class.